src/controllers/routers/referees.js

```diff
@@ -0,0 +1,84 @@
+/*!
+ * OneSG API Server by TL Bootcamp#6 OneSG Team
+ * Copyright(c) 2020 TechLadies
+ * MIT Licensed
+ */
+
+'use strict';
+
+const { check, validationResult } = require('express-validator');
+
+const {
+  errors: { BadRequest, UnprocessableEntity },
+} = require('../../utils');
+
+const db = require('../../../config/db');
+
+/**
+ * Retrieve all references
+ * @param {Request} req
+ * @param {Response} res
+ */
+const getAll = async (req, res) => {
+  const result = await db('referees');
+  res.status(200).json(result);
+};
+
+/**
+ * Create new reference
+ * @param {Request} req
+ * @param {Response} res
+ */
+const create = async (req, res) => {
+  // Return if there are any validation errors
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res
+      .status(new BadRequest().error.statusCode)
+      .json(new BadRequest(errors));
+  }
+  const newReferee = req.body;
+  try {
```

Comment:
read up on the error handling pattern here, including async:
https://expressjs.com/en/guide/error-handling.html

we should change the code to follow that pattern
the outer Promise.resolve can be written as a wrapper since all functions would need the same thing.

also, for some reason, the error handler middleware of operrors is not triggered. for now, revert to the old way of writing and invoking the errorhandler middleware.

Reply:
updated